### PR TITLE
Fix g4 issues

### PR DESCRIPTION
**Pull request by pakro**
That's my solution for the stock browser issue on the G4. By listening to all possible "transitionend" types you never miss one, and since only the one that is appropriate for the device/browser is fired, it's just the initial overhead in binding the events. But I'm honest. At this point I did only basic testing. The other part of the fix removes the proxy element that is created for listening on the "transitionend" events. The proxy element somehow doesn't get the "transitionend" event. However the mapPane does. I didn't change too much at this place. Hope that helps you somehow. These changes seem to work quite well in our project.

### CODE REVIEW

**Comment by IvanSanchez** [comment]
@pakro When you make a pull request, please mention the original bug (#3832) so magical links can happen :-)

File: src/dom/DomUtil.js [hunk 1]
```diff
@@ -182,11 +182,12 @@ L.DomUtil = {
...
 	// the same for the transitionend event, in particular the Android 4.1 stock browser
 
-	var transition = L.DomUtil.TRANSITION = L.DomUtil.testProp(
+	L.DomUtil.TRANSITION = L.DomUtil.testProp(
 			['webkitTransition', 'transition', 'OTransition', 'MozTransition', 'msTransition']);
 
-	L.DomUtil.TRANSITION_END =
-			transition === 'webkitTransition' || transition === 'OTransition' ? transition + 'End' : 'transitionend';
+	// trying to decide which type is appropriate is not as robust as adding all available types
+	// although this adds overhead (e.g LG G4 stock browser uses webkitTransition but fires transitionend)
```
Comment (IvanSanchez) [inline src/dom/DomUtil.js]: What. The. Hell.
I wonder if it's worth doing a `L.Browser` hack to detect this particular case, or we can live with the extra bit of overhead.

**Comment by IvanSanchez** [comment]
@pakro Nice work noticing the browser weirdness! I know debugging these things can be hard.

Regarding the `transitionend` events... some browsers might fire one `transitionend` on the map **and** on the proxy, or just a `transitionend` on **either** the map or the proxy, or (worst case) none. So I'm undecided about that.

**Comment by pakro** [comment]
@IvanSanchez Sorry for not linking the bug. This is actually my first pull request. Hope you can excuse this mistake. Won't happen again. :) Regarding the detection of the appropriate event. I'm not sure whether it is possible to detect the LG G4 stock browser with the user agent or not. It would be also interesting whether other LG devices (e.g G2 G3) are also affected and share the same user agent or not. At our office we only have the G4. 

Concerning the map and proxy topic. Here I honestly just made it work and tested whether it affects other devices and browser lying around our office or not. It seemed to work well for this particular selection in our office. However if you say that they are cases where it may fire either on the map or the proxy, the fix cripples down the devices/browsers where the proxy is the receiver and not the map, as I removed it. So this may be not the best solution. I'm not mad on you if you are undecided on this. I also found it very weird and didn't understand the problem in its full complexity.

I'll try finding some time to investigate the map and proxy issue.

**Comment by Eschon** [comment]
> It would be also interesting whether other LG devices (e.g G2 G3) are also affected

I have a G3 right here if you could provide a way to reproduce the bug this should fix I can test it.

**Comment by pakro** [comment]
@Eschon It's actually quite simple, check out the the current master load the debug/map/map-mobile.html on your stock browser and zoom the map with either the zoom controls or just a pinch to zoom gesture.  If the map stops reacting after that action you have the bug.

**Comment by Eschon** [comment]
Just tested it and seems to work without problems on my G3.

**Comment by pakro** [comment]
Can you please check whether your stock browser version matches that one of the LG G4. Under Settings/Apps/All/Internet what version is displayed? The LG G4 has the version 5.2.38F27. If the versions are the same it would be very weird, but if they are different that makes sense.

**Comment by Eschon** [comment]
@pakro My version is 5.2.34B13

**Comment by pakro** [comment]
OK the version is different and therefore it is very likely that the used engine has also a different version. That may a justification for a L.Browser hack, as this problem is LG G4 Android 5.1 exclusive. I assume the G2 does also use a different stock browser version.

**Comment by IvanSanchez** [comment]
This has merge conflicts, so I'll close it down for now.

I'm not very optimistic about making changes in the `animationend` event detection, because browsers must handle some (documented) strange behaviour. In other words, things can get messy.

See:
* https://bugzilla.mozilla.org/show_bug.cgi?id=1236979
* https://bugzilla.mozilla.org/show_bug.cgi?id=1236930
* https://github.com/whatwg/compat/issues/24

Updated web browsers *should* handle the compatibility issue in the `whatwg` compat list. Please reopen the bug if it's still reproducible with up-to-date browsers.

**Pull request closed without merging**